server/container_create.go

```diff
@@ -165,7 +166,10 @@ func addImageVolumes(ctx context.Context, rootfs string, s *Server, containerInf
 		}
 		switch s.config.ImageVolumes {
 		case config.ImageVolumesMkdir:
-			if err1 := os.MkdirAll(fp, 0755); err1 != nil {
+			uid := specgen.Config.Process.User.UID
+			gid := specgen.Config.Process.User.GID
+			IDs := idtools.IDPair{UID: int(uid), GID: int(gid)}
```

Comment:
nit: you can do all this in one line `IDs := idtools.IDPair{UID: int(specgen.Config.Process.User.UID), GID: int(specgen.Config.Process.User.GID)}`